ipam/manager_ipv6Ipam_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 Microsoft. All rights reserved.
+// MIT License
+
+package ipam
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-container-networking/common"
+)
+
+var (
+
+	// Pools and addresses used by tests.
+	ipv6subnet1 = "ace:cab:deca:deed::/126"
+	ipv6addr1   = "ace:cab:deca:deed::1"
+	ipv6addr2   = "ace:cab:deca:deed::2"
+	ipv6addr3   = "ace:cab:deca:deed::3"
+)
+
+func createTestIpv6AddressManager() (AddressManager, error) {
+	var config common.PluginConfig
+	var options map[string]interface{}
+
+	options = make(map[string]interface{})
+	options[common.OptEnvironment] = common.OptEnvironmentIPv6Ipam
+
+	am, err := NewAddressManager()
+	if err != nil {
+		return nil, err
+	}
+
+	err = am.Initialize(&config, options)
+	if err != nil {
+		return nil, err
+	}
+
+	amImpl := am.(*addressManager)
+	src := amImpl.source.(*ipv6IpamSource)
+	src.nodeHostname = testNodeName
+	src.subnetMaskSizeLimit = testSubnetSize
+	src.kubeClient = newKubernetesTestClient()
+
+	return am, nil
+}
+
+//
+// Address manager tests.
+//
+// Tests address spaces are created and queried correctly.
+func TestIPv6GetAddressPoolAndAddress(t *testing.T) {
```

Comment:
need separate unit test for
1. requestpool, releasepool -> releasing pool should fail if any address in that pool is used
2. requestaddress, releaseaddress

Reply:
1 - discussed offline, not a scenario
2 - updated